### DISCUSSION TITLE
How to deploy static files on OpenLiteSpeed server

### DISCUSSION THREAD

**Discussion by ChrisGV04** [antfu/vite-ssg#170] [Q&A]
Hello everyone!
I have a small problem that I can't seem to solve but I guess it should be an easy one.

I created my Vue app using vite-ssg and generated the static files using `npm run build`. Then I uploaded the dist folder into my shared hosting server and copy all of the files inside the `dist` into the `public_html`.

Until here everything is working as expected. When I enter `mydomain.com` the Vue app renders normally and all of the routes do too when I navigate within the app itself. However, when I try to enter `mydomain.com/about` from the URL search bar in the browser or reload the page, **my server throws a 404 error even if the about.html file exists in the static files** that where generated by vite-ssg.

Can someone please help me solve this???
I'm guessing it has something to do with the virtual host settings (vHost in OpenLiteSpeed) because the vHost config file was written for WordPress, since it is in a shared hosting platform. I don't know if there are some optimal vHost settings for these kind of apps built with vite-ssg. **I'm using OpenLiteSpeed and Cyberpanel as my server.**

Thank you very much for your help!

**Comment by sibbng** [comment]
Try creating `.htaccess` file on public_html directory with this content:

```
RewriteEngine On
RewriteBase /

RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule ^(.*)$ $1.html [L]
```

It might not work, haven't tested.

**Reply by ChrisGV04** [reply]
Hi! Thank you for your answer.
I tried your suggestion and it didn't work either. For some reason the `/about` and every other nested route still throw 404 error. For now, I had to set that 404 errors are rewritten to `/` and that way it works.
I'm guessing that this solution has an impact on SEO, but thankfully it is not a priority for now

**Reply by Qmlio** [reply]
Am facing the exact issue...how did you **set that 404 errors are rewritten to /** ?

**Reply by ChrisGV04** [reply]
@Qmlio I haven't needed this in a few years, but if I recall correctly, I added this to my website's vHost Conf:

```
errorpage 404 {
  url                     /
}
```

**Comment by Qmlio** [comment]
Thanks ..for the response, i will try that
